tools/rosbash/scripts/findpath.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import print_function
+import sys
+import os
+import subprocess
+
+ROS_CACHE_TIMEOUT_ENV_NAME = 'ROS_CACHE_TIMEOUT'
+ROS_LOCATIONS_ENV_NAME = 'ROS_LOCATIONS'
+ROS_LOCATION_SEP = ';'
+ERROR_PREFIX = 'Error: '
+
+def ros_location_find(package_name):
+    # process ROS_LOCATION and return if found
+    ros_location = os.environ.get(ROS_LOCATIONS_ENV_NAME)
+    if ros_location is not None:
+        locations = ros_location.split(ROS_LOCATION_SEP)
+        for loc in locations:
+            index = loc.find('=')
+            if index != -1:
+                if package_name == loc[:index]:
+                    return 0, loc[index + 1:]
+    
+    if package_name == 'log':
+        p = subprocess.Popen('roslaunch-logs', stdout=subprocess.PIPE)
+        result_location = p.communicate()[0].strip()
+        result_code = p.returncode
+        return result_code, result_location if result_code == 0 else package_name + ' not found'
+
+    if package_name == 'test_results':
+        p = subprocess.Popen('rosrun.bat rosunit test_results_dir.py', stdout=subprocess.PIPE)
+        result_location = p.communicate()[0].strip()
+        result_code = p.returncode
+        return result_code, result_location if result_code == 0 else package_name + ' not found'
+
+    # process package_name and return
+    env = os.environ
+    env[ROS_CACHE_TIMEOUT_ENV_NAME]='-1.0'
+    p = subprocess.Popen(['rospack', 'find', package_name], stdout=subprocess.PIPE)
+    result_location = p.communicate()[0].strip()
+    result_code = p.returncode
+    if result_code == 0:
+        return result_code, result_location
+
+    p = subprocess.Popen(['rosstack', 'find', package_name], stdout=subprocess.PIPE)
+    result_location = p.communicate()[0].strip()
+    result_code = p.returncode
+    if result_code == 0:
+        return result_code, result_location
+    return result_code, 'No such package/stack ' + package_name
+
+# takes as argument either just a package-path or just a pkgname
+# returns 0 for no argument or if package (+ path) exist, 1 else
+# on success with arguments print result_path or Error: error message
+def findpathmain(argv):
+    reldir = ''
+    parameters = os.path.normpath(argv[0]).split(os.path.sep)
+    package_name = parameters[0]
+    if len(parameters) > 1:
+        reldir = os.path.sep.join(parameters[1:len(parameters)])
+    else:
+        if len(argv) <2 or argv[1] != 'forceeval':
+            print (ERROR_PREFIX + package_name)
```

Comment:
Please remove the space before the open parenthesis. The error message `Error: <pkgname>` doesn't seem very helpful. Also error messages should be printed to `stderr`.

Same below.

Reply:
thanks for pointing this out, updated to print to `stderr`